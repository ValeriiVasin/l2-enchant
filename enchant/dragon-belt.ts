import { createEnchantMap } from '../helpers/create-enchant-map';
import { Item } from '../items';
import { EnchantStrategy } from '../types';

const chances = [95, 90, 85, 55, 50, 40, 33, 30, 25, 15];
const items = [
  Item.DragonBelt,
  Item.DragonBelt_1,
  Item.DragonBelt_2,
  Item.DragonBelt_3,
  Item.DragonBelt_4,
  Item.DragonBelt_5,
  Item.DragonBelt_6,
  Item.DragonBelt_7,
  Item.DragonBelt_8,
  Item.DragonBelt_9,
  Item.DragonBelt_10,
];

export const dragonBeltEnchantMap = createEnchantMap({
  items,
  chances,
  strategy: EnchantStrategy.Destroy,
});
