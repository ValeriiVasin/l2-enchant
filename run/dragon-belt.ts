import { enchant } from '../helpers/enchant';
import { header } from '../helpers/header';
import { Item } from '../items';

const times = 10_000;

console.log(header('Пояс Дракона +0 => +4'));
console.log(enchant(Item.DragonBelt, Item.DragonBelt_4, times));
console.log(header('Пояс Дракона +0 => +5'));
console.log(enchant(Item.DragonBelt, Item.DragonBelt_5, times));
console.log(header('Пояс Дракона +0 => +6'));
console.log(enchant(Item.DragonBelt, Item.DragonBelt_6, times));
console.log(header('Пояс Дракона +0 => +7'));
console.log(enchant(Item.DragonBelt, Item.DragonBelt_7, times));
console.log(header('Пояс Дракона +0 => +8'));
console.log(enchant(Item.DragonBelt, Item.DragonBelt_8, times));
console.log(header('Пояс Дракона +0 => +9'));
console.log(enchant(Item.DragonBelt, Item.DragonBelt_9, times));
console.log(header('Пояс Дракона +0 => +10'));
console.log(enchant(Item.DragonBelt, Item.DragonBelt_10, times));
