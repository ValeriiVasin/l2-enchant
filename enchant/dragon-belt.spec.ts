import { dragonBeltEnchantMap } from './dragon-belt';

it('enchant map: dragon belt', () => {
  expect(dragonBeltEnchantMap).toMatchSnapshot();
});
