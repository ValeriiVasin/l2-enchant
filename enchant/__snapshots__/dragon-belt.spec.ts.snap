// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`enchant map: dragon belt 1`] = `
Map {
  "Пояс Дракона" => Object {
    "cost": "Nothing",
    "item": "Пояс Дракона",
    "results": Array [
      Object {
        "chance": 95,
        "item": "Пояс Дракона +1",
      },
      Object {
        "chance": 5,
        "item": "Nothing",
      },
    ],
  },
  "Пояс Дракона +1" => Object {
    "cost": "Nothing",
    "item": "Пояс Дракона +1",
    "results": Array [
      Object {
        "chance": 90,
        "item": "Пояс Дракона +2",
      },
      Object {
        "chance": 10,
        "item": "Nothing",
      },
    ],
  },
  "Пояс Дракона +2" => Object {
    "cost": "Nothing",
    "item": "Пояс Дракона +2",
    "results": Array [
      Object {
        "chance": 85,
        "item": "Пояс Дракона +3",
      },
      Object {
        "chance": 15,
        "item": "Nothing",
      },
    ],
  },
  "Пояс Дракона +3" => Object {
    "cost": "Nothing",
    "item": "Пояс Дракона +3",
    "results": Array [
      Object {
        "chance": 55,
        "item": "Пояс Дракона +4",
      },
      Object {
        "chance": 45,
        "item": "Nothing",
      },
    ],
  },
  "Пояс Дракона +4" => Object {
    "cost": "Nothing",
    "item": "Пояс Дракона +4",
    "results": Array [
      Object {
        "chance": 50,
        "item": "Пояс Дракона +5",
      },
      Object {
        "chance": 50,
        "item": "Nothing",
      },
    ],
  },
  "Пояс Дракона +5" => Object {
    "cost": "Nothing",
    "item": "Пояс Дракона +5",
    "results": Array [
      Object {
        "chance": 40,
        "item": "Пояс Дракона +6",
      },
      Object {
        "chance": 60,
        "item": "Nothing",
      },
    ],
  },
  "Пояс Дракона +6" => Object {
    "cost": "Nothing",
    "item": "Пояс Дракона +6",
    "results": Array [
      Object {
        "chance": 33,
        "item": "Пояс Дракона +7",
      },
      Object {
        "chance": 67,
        "item": "Nothing",
      },
    ],
  },
  "Пояс Дракона +7" => Object {
    "cost": "Nothing",
    "item": "Пояс Дракона +7",
    "results": Array [
      Object {
        "chance": 30,
        "item": "Пояс Дракона +8",
      },
      Object {
        "chance": 70,
        "item": "Nothing",
      },
    ],
  },
  "Пояс Дракона +8" => Object {
    "cost": "Nothing",
    "item": "Пояс Дракона +8",
    "results": Array [
      Object {
        "chance": 25,
        "item": "Пояс Дракона +9",
      },
      Object {
        "chance": 75,
        "item": "Nothing",
      },
    ],
  },
  "Пояс Дракона +9" => Object {
    "cost": "Nothing",
    "item": "Пояс Дракона +9",
    "results": Array [
      Object {
        "chance": 15,
        "item": "Пояс Дракона +10",
      },
      Object {
        "chance": 85,
        "item": "Nothing",
      },
    ],
  },
}
`;
